### ISSUE TITLE
Better cleanup in case of exceptions and program exit for the lux chaos

### ISSUE THREAD

**Issue by cwebber** [jeapostrophe/raart#5]
When a program exits, intentionally or exceptionally, my terminal never looks good.  I have some code to make it a bit less bad which has mostly been guesswork from reading the `ansi` source code:

```racket
(define (with-clean-tty proc)
  (tty-raw!)
  (display* (dec-soft-terminal-reset)
            (device-request-screen-size))
  (match-define (screen-size-report rows columns)
    (lex-lcd-input (current-input-port)))
  (define proc-result
    (proc rows columns))
  (display* (dec-soft-terminal-reset)
            (clear-screen/home))
  proc-result)

(define (start-game)
  (with-clean-tty
    (lambda (rows cols)
      (call-with-chaos
       (raart:make-raart)
       (lambda ()
         (fiat-lux (new-game rows cols)))))))
```

This kind of works for intentional exit; it doesn't restore my cursor but otherwise seems fine.

However, in the case of "unintentional exit", exceptions still look like the following (this one taken from Jack the Paren Hunter example; press any key other than the movement keys):

```
match: no matching clause for "C-M"
                                                     location...:
                                                                    devel/raart/t/hack.rkt:47:5
                                                                                                 context...:
                                                                                                               /gnu/store/5vx08s9v4rv1lk4h0aq1kn4kb3d6v6xg-racket-7.3/share/racket/collects/racket/match/runtime.rkt:24:0: match:error
                                                                           /home/cwebber/.racket/7.3/pkgs/lux/word.rkt:157:18
                                                                                                                                /home/cwebber/.racket/7.3/pkgs/lux/word.rkt:84:0: call-with-chaos
                                      (submod "/home/cwebber/devel/raart/t/hack.rkt" main): [running body]
                                                                                                             temp37_0
                                                                                                                        for-loop
                                                                                                                                   run-module-instance!125
```

Everything's moving to the right!  It's hard to follow.

I'd like to have something better; what, I'm not sure.  Thoughts?  What can we do?

**Comment by jeapostrophe** [comment]
There's a better fix for the total screwup in lux, which I just pushed. As for the everything moving to the right, I'm not sure what's going on there...
